Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -536,48 +552,57 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         do {
             // try to get it from the cache
             if let result = try cache?.get(key: key.sha256Checksum), let manifestJSON = result.manifestJSON, !manifestJSON.isEmpty {
-                return try self.parseManifest(
+                return completion(.success(try self.parseManifest(
                     result,
                     packageIdentity: packageIdentity,
                     packageKind: packageKind,
                     toolsVersion: toolsVersion,
                     identityResolver: identityResolver,
                     fileSystem: fileSystem,
                     observabilityScope: observabilityScope
-                )
+                )))
             }
         } catch {
             observabilityScope.emit(warning: "failed loading cached manifest for '\(key.packageIdentity)': \(error)")
+            completion(.failure(error))
         }
 
         // shells out and compiles the manifest, finally output a JSON
-        let result = self.evaluateManifest(
+        self.evaluateManifest(
             packageIdentity: key.packageIdentity,
             manifestPath: key.manifestPath,
             manifestContents: key.manifestContents,
             toolsVersion: key.toolsVersion,
             delegateQueue: delegateQueue
-        )
+        ) { result in
+            do {
+                let evaluationResult : EvaluationResult
+                switch result {
+                case .success(let result):
+                    evaluationResult = result
+                case .failure(let error):
+                    return completion(.failure(error))
+                }
```

Comment:
similar to https://github.com/apple/swift-package-manager/pull/3872/files#r750486345

Reply:
Fixed in fd4f7e27.